### PR TITLE
Add Dart AOT integration test

### PR DESCRIPTION
**Pull request by lin-erik**
This PR introduces a few changes to the existing Dart JIT integration test and adds the new Dart AOT integration test.

- There is now a `dart_echo_server` directory that contains both versions of the Echo server needed by the JIT/AOT integration tests. The versions will match the runner required by their respective test.
- Integration tests now use `oot_dart_*_runner`. This is just a copy of the `dart_*_runner` but with a custom name. This will avoid conflicting with the Dart runners used in the base Fuchsia image. Changes made to the Dart runner are now picked up as expected.
- Integration tests launch the Echo server component through the RealmBuilder, with a specified runner and necessary capabilities

### CODE REVIEW

**Comment by zanderso** [comment]
From PR review triage: Marking as WIP/Draft.

**Comment by skia-gold** [comment]
Gold has detected about 7 new digest(s) on patchset 20.
View them at https://flutter-engine-gold.skia.org/cl/github/36460

File: shell/platform/fuchsia/dart_runner/tests/startup_integration_test/dart_jit_runner/dart-jit-runner-integration-test.cc [hunk 1]
```diff
@@ -73,10 +75,15 @@ TEST_F(RealmBuilderTest, DartRunnerStartsUp) {
...
+  // Route base capabilities to the Dart JIT runner
   realm_builder.AddRoute(
-      Route{.capabilities = {Protocol{"fuchsia.logger.LogSink"}},
+      Route{.capabilities = {Protocol{"fuchsia.logger.LogSink"},
+                             Protocol{"fuchsia.tracing.provider.Registry"},
+                             Protocol{"fuchsia.posix.socket.Provider"},
+                             Protocol{"fuchsia.intl.PropertyProvider"}},
             .source = ParentRef(),
-            .targets = {kDartJitEchoServerRef}});
+            .targets = {kDartJitRunnerRef}});
```
Comment (richkadel) [inline shell/platform/fuchsia/dart_runner/tests/startup_integration_test/dart_jit_runner/dart-jit-runner-integration-test.cc:86]: You should still also route LogSink to the EchoServer (in addition to routing LogSink to the runner).

File: shell/platform/fuchsia/dart_runner/tests/startup_integration_test/dart_echo_server/README.md [hunk 2]
```diff
@@ -0,0 +1,7 @@
+# dart_echo_server
+
+Contains the Echo servers for the integration tests. There are currently two outputs:
+- `dart-aot-echo-server`
+- `dart-jit-ehco-server`
```
Comment (akbiggs) [inline shell/platform/fuchsia/dart_runner/tests/startup_integration_test/dart_echo_server/README.md:5]: nit: echo

**Review by akbiggs — APPROVED** [review]
Awesome!

**Pull request merged**